lib/model.js

```diff
@@ -0,0 +1,233 @@
+    concat = require('concat-stream'),
+    util = require('util'),
+    EventEmitter = require('events').EventEmitter;
+
+var Model = function (attributes, options) {
+
+    this.options = options || {};
+    this.attributes = {};
+    this.set(attributes, { silent: true });
+};
+
+util.inherits(Model, EventEmitter);
+
+_.extend(Model.prototype, {
+
+    save: function (options, callback) {
+        if (typeof options === 'function' && arguments.length === 1) {
+            callback = options;
+            options = {};
+        }
+        this.prepare(function (err, data) {
+
+            if (err) { return callback(err); }
+
+            data = JSON.stringify(data);
+
+            var reqConf = url.parse(this.url(options));
+            reqConf.method = options.method || 'POST';
+
+            reqConf.headers = {
+                'Content-Type': 'application/json',
+                'Content-Length': data.length
+            };
+
+            reqConf.data = data;
+
+            this.request(reqConf, callback);
+
+        }.bind(this));
+    },
+
+    fetch: function (options, callback) {
+        if (typeof options === 'function' && arguments.length === 1) {
+            callback = options;
+            options = {};
+        }
+        var reqConf = url.parse(this.url(options));
+        reqConf.method = options.method || 'GET';
+        this.request(reqConf, callback);
+    },
+
+    delete: function (options, callback) {
+        if (typeof options === 'function' && arguments.length === 1) {
+            callback = options;
+            options = {};
+        }
+        var reqConf = url.parse(this.url(options));
+        reqConf.method = options.method || 'DELETE';
+        this.request(reqConf, callback);
+    },
+
+    request: function (settings, callback) {
+        var protocol = (settings.protocol === 'http:') ? http : https;
+        settings.auth = this.auth();
+
+        var _callback = function (err, data, statusCode) {
+            if (err) {
+                this.emit('fail', err, data, settings, statusCode);
+            } else {
+                this.emit('success', data, settings, statusCode);
+            }
+            callback(err, data);
+        }.bind(this);
+
+        var request = protocol.request(settings, function (response) {
+            this.handleResponse(response, settings, _callback);
+        }.bind(this));
+        request.on('error', function(e) {
+            _callback(e);
+        });
+        this.emit('sync', settings);
+        if (settings.data) {
+            request.write(settings.data);
+        }
+        request.end();
+
+    },
+
+    handleResponse: function(response, settings, callback) {
+
+        response.pipe(concat(function (d) {
+            var data = {};
+            try {
+                data = JSON.parse(d.toString() || '{}');
+            } catch (e) {
+                e.status = response.statusCode;
+                return callback(e, null, response.statusCode);
```

Comment:
Since I appear to have knackered the diff by moving a file, this is where the change is.